src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Deserialize.cs

```diff
@@ -1341,7 +1341,7 @@ public async Task DuplicatedIdArray()
 
             JsonException ex = await Assert.ThrowsAsync<JsonException>(async () => await Serializer.DeserializeWrapper<ClassWithTwoListProperties>(json, s_deserializerOptionsPreserve));
 
-            Assert.Equal("$.List2.$id", ex.Path);
+            Assert.Equal("$.List2.$values", ex.Path);
```

Comment:
Delayed evaluation of `$id` means that the error will be hit when attempting to create the collection (which is after the first token inside `$values` has been read).